extensions/amp-cxense-player/0.1/amp-cxense-player.js

```diff
@@ -0,0 +1,244 @@
+    }
+
+    /** @override */
+    isLayoutSupported(layout) {
+      return isLayoutSizeDefined(layout);
+    }
+
+    /** @override */
+    renderOutsideViewport() {
+      return false;
+    }
+
+    /** @override */
+    buildCallback() {
+      this.element.classList.add('amp-cxense-player');
+
+        /** @private @const {object} */
+      this.cxDefaults_ = JSON.parse(JSON.stringify(cxDefaults));
+        /** @private @const {object} */
+      this.attrs_ = this.getDataAttributes_();
+
+      this.cxDefaults_.embedHost = this.attrs_.debug
+            ? this.cxDefaults_.debugEmbedHost : this.cxDefaults_.distEmbedHost;
+      this.cxDefaults_.embedApp = this.attrs_.debug
+            ? this.cxDefaults_.debugEmbedApp : this.cxDefaults_.distEmbedApp;
+
+        /** @private @const {number} */
+      this.width_ = getLengthNumeral(this.element.getAttribute('width'));
+        /** @private @const {number} */
+      this.height_ = getLengthNumeral(this.element.getAttribute('height'));
+
+      if (!this.getPlaceholder()) {
+        this.buildImagePoster_();
+        this.loading_(true);
+      }
+    }
+
+    /** @override */
+    layoutCallback() {
+      const self = this;
+
+        /** @private @const {Element} */
+      this.iframe_ = this.element.ownerDocument.createElement('iframe');
+
+      this.iframe_.setAttribute('frameborder', '0');
+      this.iframe_.setAttribute('allowfullscreen', 'true');
+
+      this.iframe_.width = this.width_;
+      this.iframe_.height = this.height_;
+
+      this.iframe_.src = this.getIframeSrc_();
+      setStyles(this.iframe_, {display: 'none'});
+      this.element.appendChild(this.iframe_);
+
+        /** @private {boolean} */
+      this.mpfReady_ = false;
+
+        /** @private @const {!Promise} */
+      this.playerReadyPromise_ = new Promise(resolve => {
+            /** @private @const {function()} */
+        this.playerReadyResolver_ = iframe => {
+          self.mpfReady = true;
+          resolve(iframe);
+        };
+      });
+
+      this.handleFrameMessages_ = this.handleFrameMessages_.bind(this);
+      this.getWin().addEventListener('message', this.handleFrameMessages_);
+
+      return loadPromise(this.iframe_)
+            .then(() => this.playerReadyPromise_)
+            .then(ret => {
+              self.loading_(false);
+              setStyles(self.iframe_, {display: ''});
+              self.applyFillContent(self.iframe_);
+              return ret;
+            });
+    }
+
+    /** @override */
+    pauseCallback() {
+      this.postMessage_({type: 'mpf.video.pause'});
+      return true;
+    }
+
+    /** @override */
+    unlayoutCallback() {
+      this.getWin().removeEventListener('message', this.handleFrameMessages_);
+      this.iframe_.setAttribute('src', 'about:blank');
+      this.iframe_.parentNode.removeChild(this.iframe_);
+      return true;
+    }
+
+    /** @private */
+    handleFrameMessages_(event) {
```

Comment:
We have [`listenFor`](https://github.com/ampproject/amphtml/blob/7b2c9b9be2eaa4eb4433f4e647d3f0d18d3735cc/src/iframe-helper.js#L230-L293) and `listenForOnce` that will do exactly this.